### DISCUSSION TITLE
Spleen output segmentation in different orientation

### DISCUSSION THREAD

**Discussion by park-minjun** [Project-MONAI/tutorials#1113] [Q&A]
Hello,

I've wanted to run through the spleen-tutorial and decided to copy-paste the entire code given to train and output the test-dataset segmentation. However, when I get the segmentation file for the test dataset and view it in 3D Slicer, the output segmentation is in a different orientation than the original volume, leading to a mismatch. The matplotlib shows it aligning well, so I was wondering if anyone has had an issue like this previously and found a solution for this?

Previous searches of a similar issue end with an incomplete resolution where no specific source of error was pin-pointed. Also, this error is directly from the "Spleen 3D segmentation with MONAI" tutorial, so I'm not sure where to start. Thanks!

**Comment by diazandr3s** [comment]
Hi @park-minjun,

This issue comes from two sources: predictions aren't oriented back to the original or the loader missinterpreted file headers.

Make sure the post transforms inverted the orientation back to the original ones.

Which file format are the images you use? You may also want to check a different type of loader when using the `LoadImaged(keys="image")` transform.

Something like: 

`LoadImaged(keys=("image", "label"), reader='itkreader')`

or

`LoadImaged(keys=("image", "label"), reader='NibabelReader')`

**Reply by park-minjun** [reply]
Hello, thank you for your response.

I have tried adding those different parameters but it seems like it's not working. What is weird is I'm copying and pasting from the tutorial provided by MONAI but it is having these issues. I am wondering if maybe the post transformations are not probably working properly anymore due to some updates? I recall this working couple of months back when copying and pasting from the tutorial, now it's not working anymore.

Does anyone have insight into this?

**Reply by park-minjun** [reply]
Also, to be more clear, the InvertD seems to work fine in validation transformation in the tutorial. It is just the outputting the nii.gz file that causes these errors
